src/components/inputs/SupplierField.tsx

```diff
@@ -3,23 +3,29 @@ import searchIcon from '../icons/searchIcon';
 import closeIcon from '../icons/closeIcon';
 import { ChangeEvent } from 'react';
 import { Textfield } from '../base/Textfield';
-import '../../styles/newCertificate.css';
+import { Supplier } from '../data/data';
+import '../../styles/certificateForm.css';
 
 interface SupplierFieldProps {
-  supplier: string;
-  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  supplier: Supplier;
+  onChange: (supplier: Supplier) => void;
 }
 
 export function SupplierField(props: SupplierFieldProps) {
+  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    props.onChange({ ...props.supplier, name: value });
```

Comment:
I don't think it's smart idea of merging old value with new one. Here you select new Supplier, so new one should be set, not merging name of new one with old data you had. Especially because this will be changed completelly with more data